schema/schema_test.go

```diff
@@ -0,0 +1,389 @@
+package schema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCollectionSchema(t *testing.T) {
+	// invalid
+	type empty struct {
+	}
+
+	// invalid
+	type noPK struct {
+		Key1 string `json:"key_1"`
+	}
+
+	// valid. implicit one part primary key with index 1
+	type pk struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key"`
+	}
+
+	// invalid. index starts from 1
+	type pk0 struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key:0"`
+	}
+
+	// valid. optional index specified
+	type pk1 struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// valid. composite primary key
+	type pk2 struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:2"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	type pkDup struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:1"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// invalid. promary key index greater then number of fields
+	type pkOutOfBound struct {
+		Key3 string `json:"key_3" tigrisdb:"primary_key:3"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// invalid. gap in primary key indexes
+	type pkGap struct {
+		Key3  string `json:"key_3" tigrisdb:"primary_key:3"`
+		Key1  string `json:"key_1" tigrisdb:"primary_key:1"`
+		Data1 string
+	}
+
+	type pkInvalidType struct {
+		Key2 bool `json:"key_2" tigrisdb:"primary_key:1"`
+	}
+
+	type pkInvalidTag struct {
+		Key2 bool `json:"key_2" tigrisdb:"primary_key:1:a"`
+	}
+
+	type pk3 struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:2"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+		Key3 string `json:"key_3" tigrisdb:"primary_key:3"`
+	}
+
+	type subSubStructPK struct {
+		Data1  int    `tigrisdb:"-"`
```

Comment:
what is this tag means here `tigrisdb:"-"`

Reply:
Skip this field. Same as JSON tag meaning.